src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -128,6 +129,74 @@ public static async Task ZipArchiveEntry_InvalidUpdate(string zipname)
             }
         }
 
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Fix not shipped for .NET Framework.")]
+        public static void ZipArchiveEntry_BypassValidationWhenSwitchOn()
+        {
+            RemoteExecutor.Invoke(() =>
+            {
+                MemoryStream stream = populateStream().Result;
+                PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(s_tamperedFileName), stream, 8);  // patch uncompressed size in file header
+                using (ZipArchive archive = new ZipArchive(stream, ZipArchiveMode.Read))
+                {
+                    // Set the AppContext Switch
+                    AppContext.SetSwitch("System.Compression.ZipArchiveEntry.SuppressHeaderValidation", true);
+                    ZipArchiveEntry e = archive.GetEntry(s_tamperedFileName);
+
+                    using (MemoryStream ms = new MemoryStream())
+                    using (Stream source = e.Open())
+                    {
+                        source.CopyTo(ms);
+                        Assert.Equal(ms.Position, ms.Length);  // Just making sure it was readable
+                    }
+                }
+            }).Dispose();
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Fix not shipped for .NET Framework.")]
+        public static void ZipArchiveEntry_ValidationThrowsWhenSwitchOff()
```

Comment:
(minor) thoughts on using a `[Theory]` here passing in `true` and `false`. That way we can ensure the same "patch data" happens in both cases. With the logic duplicated, it may get changed in one place, but not the other.

Reply:
yep, makes sense, thanks

Reply:
Seems we cannot use parameterized test with RemoteExecutor

> ZipArchiveEntry_BypassValidationWhenSwitchOn(suppressValidation: True) [FAIL]
>         Assert.All() Failure: 1 out of 1 items in the collection did not pass.
>         [0]: Item: Boolean suppressValidation
>              Xunit.Sdk.TrueException: Field marshaling is not supported by Invoke: suppressValidation

Reply:
`RemoteExecutor.Invoke` can take an `Action<string>`, so you can pass `"true"` or `"false"` into the remote method.

```C#
        [Theory]
        [InlineData(true)]
        [InlineData(false)]
        public static void ZipArchiveEntry_ValidationThrowsWhenSwitchOn(bool suppressParam)
        {
            RemoteExecutor.Invoke(suppressString =>
            {
                bool suppress = bool.Parse(suppressString);
...
            }, suppressParam.ToString()).Dispose();
        }
```
Or you could have 2 `Fact` methods each calling RemoteExecutor, and then inside the delegate they invoke a common method with a hard-coded `true` or `false`. Ex:

```C#
        [Fact]
        public static void ZipArchiveEntry_ValidationThrowsWhenSwitchOn()
        {
            RemoteExecutor.Invoke(() =>
            {
                CommonMethod(true);
            }).Dispose();
        }

        [Fact]
        public static void ZipArchiveEntry_ValidationThrowsWhenSwitchOff()
        {
            RemoteExecutor.Invoke(() =>
            {
                CommonMethod(false);
            }).Dispose();
        }

        private static void CommonMethod(bool value) {... }
```